File: TT02/Palindromo/Palindromo.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

#define NUMLINHA    1000
#define TAMPALAVRA  100

//String s      -> char s[]
//s.length()==3 -> tamanho igual a 3
//strlen(s) >=3 -> tamanho menor ou igual a 3
//charAt(i)     -> s[0]
bool isFim(char s[]){
        return (strlen(s)>=3 && s[0]=='F' && s[1]=='I' && s[2]=='M');
}


bool isPalindromo(char s[]){
    //s.length() -> strlen
    int last = strlen(s)-1;
    //
    for(int i=0; i<last; i++, last--){
        //s.charAt() -> s[posicao]
        if(s[i] != s[last]){
            return false;
        }
    }
    return true;
}

int main(){
    //String[] entrada = new String [1000];
    char entrada[NUMLINHA][TAMPALAVRA];
    int numEntrada=0;
    //
    do{
        //entrada[numEntrada]=MyIO.readLine();
        // scanf("%[^\n]%*c" -> ??????
        scanf("%[^\n]%*c", entrada[numEntrada]);
        //Desconsidera a linha da palavra FIM
        if(isFim(entrada[numEntrada])==false){
            //Se palindromo Print SIM/NAO
            if (isPalindromo(entrada[numEntrada]) == true)
                    printf("SIM\n");
                else
                    printf("NAO\n");
        }
    } while (isFim(entrada[numEntrada++])==false);
    numEntrada--;   //Desconsidera a linha da palavra FIM
}

File: TTs/TT02/Palindromo/Palindromo.java
/*

Faça um programa recebe várias frases e identifica se elas são palíndromos
Entrada : composta por várias linhas sendo que a última contém a palavra FIM
Saída: para cada linha de entrada, escreva SIM / NÃO, indicando se a linha é um palíndromo

*/

class Palindromo{
    /*
    Utilizado para leitura de arquivo até encontrar fim
        @param  String a testar
        @return true -> fim     encontrado
                false-> fim nao encontrado
    */
    public static boolean isFim(String s){
        return (s.length()==3 && s.charAt(0)=='F' && s.charAt(1)=='I' && s.charAt(2)=='M');
    }

    /*
    Testa se Palindromo
        @param  String a testar
        @return true -> se     for palindromo
                false-> se nao for palindromo
    */
    public static boolean isPalindromo(String s){
        //last -> tamanho da palavra -1, ou seja, a ultima posicao
        int last = s.length()-1;
        //testa enquanto i menor que a ultima palavra
        for(int i=0; i<last; i++,last--){
            //testar se nao eh palindromo
            if(s.charAt(i) != s.charAt(last)){
                return false;
            }
        }
        //Deu tudo certo
        return true;
    }

    /*
    Main
    */
    public static void main (String[] args){
        //Variaveis de entrada
        String[] entrada = new String[1000];
        int numEntrada=0;
        //Leitura da entrada padrao
        // do while not FIM
        do {
            //
            entrada[numEntrada]=MyIO.readLine();
            //Desconsidera a linha da palavra FIM
            if(isFim(entrada[numEntrada])==false){
                //Se palindromo Print SIM/NAO
                if (isPalindromo(entrada[numEntrada]) == true)
                    System.out.println("SIM");
                else
                    System.out.println("NAO");
            }
        } while (isFim(entrada[numEntrada++])==false);
        numEntrada--;   //Desconsidera a linha da palavra FIM

        /*Se palindromo Print SIM/NAO
        for(int i=0; i<numEntrada; i++){
            if (isPalindromo(entrada[i]) == true)
                System.out.println("TRUE");
            else
                System.out.println("FALSE");
        */
    }
}